### PR TITLE
Use code as sample ID on sample imports

### PR DESCRIPTION
**Pull request by josh13w**
Fix [this](https://github.com/tnc-br/bugbash/issues/77) bug. When samples are imported, we now require a code value for all rows and use that value as the ID for the sample. If a sample already exists in the collection with that ID the old sample is replaced with the new data. 

Manual testing: 
1. Attempt to upload [this](https://docs.google.com/spreadsheets/d/1hm0jaDYBXzSAIXi-WpJeXB_0KO2vIuBmqAflv1jCcNw/edit#gid=409109326) csv file.
2. In the completed table search for 'mad'. Confirm that there are three samples there with the correct data. 
3. Try to upload the same file again. 
4. Confirm that the total number of samples in the database did not change (because the samples were replaced). 

Tests: `npm run test`
```
Test Suites: 8 passed, 8 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        2.083 s
Ran all test suites.
```

### CODE REVIEW

File: sample_tracking/app/i18n/locales/en/translations.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
   "download": "Download",
   "successfullyImportedFile" : "Successfully imported file",
   "great": "Great!",
-  "originValueError": "origin value should be one of the following: known, uncertain, or unknown",
+  "originValueError": "trusted value should be one of the following: known, uncertain, or unknown",
```
Comment (benwulfe) [inline sample_tracking/app/i18n/locales/en/translations.json:84]: there seem to be a lot of uses of 'origin'  should you do a full search replace?  This doesnt seem related to this change.

File: sample_tracking/app/i18n/locales/en/translations.json [hunk 2]
```diff
@@ -81,7 +81,7 @@
...
   "originValueRequired": "origin value is required, ",
```
Comment (benwulfe) [inline sample_tracking/app/i18n/locales/en/translations.json:85]: origjn appears here...

File: sample_tracking/app/import-samples.tsx [hunk 3]
```diff
@@ -234,7 +234,7 @@ export default function ImportSamples() {
                 //RFC 3339 format
                 const formattedDateString = date.toISOString();
                 Object.keys(codeList).forEach((key: string) => {
-                    const sampleId = getRanHex(20);
+                    const sampleId = key;
```
Comment (benwulfe) [inline sample_tracking/app/import-samples.tsx:237]: can you add a code comment that "code lab" is actually "code"

**Review by benwulfe — APPROVED** [review]
it looks good based on my review, because you are using batch.set to overwrite any document that exists so you should not have any carry over values.  but its worth double checking that by removing a col like species and re-importing.  species should be blank.

File: sample_tracking/app/i18n/locales/en/translations.json (hunk 1 above)
Comment (josh13w) [reply sample_tracking/app/i18n/locales/en/translations.json:84]: yeah I think a followup bug fix that would be a bit more invasive would be replacing all references of origin with trusted.

**Review by kazob1998 — APPROVED** [review]
LGTM

**Pull request merged**